packages/manager/modules/iam/src/components/advancedModeSwitch/advancedModeSwitch.controller.js

```diff
@@ -0,0 +1,26 @@
+export default class IamAdvancedModeSwitchController {
+  /* @ngInject */
+  constructor($q, $rootScope, PreferencesService) {
+    this.$q = $q;
+    this.$rootScope = $rootScope;
+    this.PreferencesService = PreferencesService;
+  }
+
+  onChangeMode(modelValue) {
+    return this.$q
+      .resolve(
+        this.PreferencesService.editPreference(
+          this.$rootScope.IAM.KEY_PREFERENCES.advancedMode,
+          `${modelValue}`,
+        ),
+      )
+      .then(() => {
+        return this.goTo({ name: '.', reload: true });
+      })
```

Comment:
Return statement can be omitted
```suggestion
      .then(() => this.goTo({ name: '.', reload: true }))
```